src/tck/src/SdkClient.cc

```diff
@@ -39,21 +57,170 @@ constexpr auto DEFAULT_TCK_REQUEST_TIMEOUT = std::chrono::seconds(30);
+  {
+    // If no type is specified, nothing else should be specified.
+    if (key.mFromKey.has_value() || key.mThreshold.has_value() || key.mKeys.has_value())
+    {
+      throw JsonRpcException(JsonErrorType::INVALID_REQUEST,
+                             "invalid request: no other parameters should be specified if type isn't specified.");
+    }
+
+    // Generate a random key.
+    switch (internal::Utilities::getRandomNumber(0, 3))
+    {
+      case 0:
+        return ED25519PrivateKey::generatePrivateKey()->toStringDer();
+      case 1:
+        return ED25519PrivateKey::generatePrivateKey()->getPublicKey()->toStringDer();
+      case 2:
+        return ECDSAsecp256k1PrivateKey::generatePrivateKey()->toStringDer();
+      case 3:
+        return ECDSAsecp256k1PrivateKey::generatePrivateKey()->getPublicKey()->toStringDer();
+      default:
+        // This should never happen.
+        throw std::range_error("Random number generated outside specified range.");
+    }
+  }
+
+  if (key.mType.value() == "ed25519PrivateKey")
+  {
+    return ED25519PrivateKey::generatePrivateKey()->toStringDer();
+  }
+
+  if (key.mType.value() == "ecdsaSecp256k1PrivateKey")
+  {
+    return ECDSAsecp256k1PrivateKey::generatePrivateKey()->toStringDer();
+  }
+
+  if (key.mType.value() == "ed25519PublicKey")
+  {
+    return (key.mFromKey.has_value())
+             ? ED25519PrivateKey::fromStringDer(key.mFromKey.value())->getPublicKey()->toStringDer()
+             : ED25519PrivateKey::generatePrivateKey()->getPublicKey()->toStringDer();
+  }
+
+  if (key.mType.value() == "ecdsaSecp256k1PublicKey")
+  {
+    return (key.mFromKey.has_value())
+             ? ECDSAsecp256k1PrivateKey::fromStringDer(key.mFromKey.value())->getPublicKey()->toStringDer()
+             : ECDSAsecp256k1PrivateKey::generatePrivateKey()->getPublicKey()->toStringDer();
+  }
+
+  if (const bool isThreshold = key.mType.value() == "thresholdKey"; key.mType.value() == "keyList" || isThreshold)
+  {
+    if (!key.mKeys.has_value())
+    {
+      throw JsonRpcException(JsonErrorType::INVALID_REQUEST,
+                             "invalid request: keys list is required for generating a KeyList type.");
+    }
+
+    if (isThreshold && !key.mThreshold.has_value())
+    {
+      throw JsonRpcException(JsonErrorType::INVALID_REQUEST,
+                             "invalid request: threshold is required for generating a ThresholdKey type.");
+    }
+
+    KeyList keyList;
+    std::for_each(key.mKeys->cbegin(),
+                  key.mKeys->cend(),
+                  [&keyList](const TckKey& key) { keyList.push_back(getKey(generateKeyHelper(key))); });
+
+    if (isThreshold)
+    {
+      keyList.setThreshold(key.mThreshold.value());
+    }
+
+    return internal::HexConverter::bytesToHex(
+      internal::Utilities::stringToByteVector(keyList.toProtobufKey()->SerializeAsString()));
+  }
+
+  if (key.mType.value() == "privateKey")
+  {
+    return internal::Utilities::getRandomNumber(0, 1) > 0
```

Comment:
nit: Can we create a function that does the whole getRandomNumber logic that returns a key?

Reply:
What kind of function are you thinking? `getRandomNumber` already hides a lot of the internals, I feel like this is already as condensed as it can be